service/db/src/test/java/gov/va/vro/service/db/SaveToDbServiceImplTest.java

```diff
@@ -40,71 +35,146 @@ class SaveToDbServiceImplTest {
+    claim.setBenefitClaimId("1234");
+    claim.setIdType(MasAutomatedClaimPayload.CLAIM_V2_ID_TYPE);
     claim.setVeteranIcn("v1");
     claim.setDiagnosticCode("7101");
     saveToDbService.insertClaim(claim);
-    ClaimEntity claimBeforeAssessment =
-        claimRepository.findByClaimSubmissionId("1234").orElseThrow();
+    ClaimEntity claimBeforeAssessment = claimRepository.findByVbmsId("1234").orElseThrow();
     Map<String, Object> evidenceMap = new HashMap<>();
     evidenceMap.put("medicationsCount", "10");
     AbdEvidenceWithSummary evidence = new AbdEvidenceWithSummary();
+    evidence.setIdType(MasAutomatedClaimPayload.CLAIM_V2_ID_TYPE);
     evidence.setEvidenceSummary(evidenceMap);
+    // evidence flag is currently null
     saveToDbService.insertAssessmentResult(claimBeforeAssessment.getId(), evidence, "7101");
-    ClaimEntity result = claimRepository.findByClaimSubmissionId("1234").orElseThrow();
+    evidence.setSufficientForFastTracking(false);
+    saveToDbService.updateSufficientEvidenceFlag(evidence, "7101");
+    ClaimEntity result = claimRepository.findByVbmsId("1234").orElseThrow();
     assertNotNull(result);
     assertNotNull(result.getContentions().get(0).getAssessmentResults().get(0));
     AssessmentResultEntity assessmentResult =
         result.getContentions().get(0).getAssessmentResults().get(0);
     assertEquals(assessmentResult.getEvidenceCountSummary(), evidenceMap);
+    assertEquals(assessmentResult.getSufficientEvidenceFlag(), false);
 
     long c = assessmentResultRepository.count();
     assertEquals(1, c);
   }
 
+  @Test
+  void persistOffRampReason() {
+    Claim claim = new Claim();
+    claim.setBenefitClaimId("1234");
+    claim.setCollectionId("collection1");
+    claim.setIdType(MasAutomatedClaimPayload.CLAIM_V2_ID_TYPE);
+    claim.setVeteranIcn("v1");
+    claim.setDiagnosticCode("7101");
+    saveToDbService.insertClaim(claim);
+    ClaimEntity result1 = claimRepository.findByVbmsId("1234").orElseThrow();
+    assertNotNull(result1);
+    Set<ClaimSubmissionEntity> csEntities = result1.getClaimSubmissions();
+    assertEquals(1, csEntities.size());
+    ClaimSubmissionEntity claimSubmission1 = csEntities.iterator().next();
+    assertNull(claimSubmission1.getOffRampReason());
+    claim.setOffRampReason("OffRampReason1");
+    saveToDbService.setOffRampReason(claim);
+    ClaimEntity result2 = claimRepository.findByVbmsId("1234").orElseThrow();
+    assertNotNull(result2);
+    Set<ClaimSubmissionEntity> csEntities2 = result2.getClaimSubmissions();
+    assertEquals(1, csEntities2.size());
+    ClaimSubmissionEntity claimSubmission2 = csEntities.iterator().next();
+    assertEquals(claim.getOffRampReason(), claimSubmission2.getOffRampReason());
+  }
+
   @Test
   void persistEvidenceSummaryDocument() throws Exception {
     // Save claim
     Claim claim = new Claim();
-    claim.setClaimSubmissionId("1234");
+    claim.setBenefitClaimId("787878");
```

Comment:
For clarity
```suggestion
    vbmsId="787878"
    claim.setBenefitClaimId(vbmsId);
```
and change line 194 below to use the `vbmsId` variable.